src/packages/recompose/flattenProps.js

```diff
@@ -0,0 +1,20 @@
+import createHelper from './createHelper'
+import createEagerFactory from './createEagerFactory'
+
+const flattenProps = propNames => BaseComponent => {
+  const factory = createEagerFactory(BaseComponent)
+
+  return props =>
+    factory({
+      ...props,
+      ...propNames.reduce(
+        (flattenedProps, propName) => ({
+          ...flattenedProps,
+          ...props[propName],
+        }),
+        {}
+      ),
```

Comment:
Since this is a "library-like" utility function, maybe it would be worth it to mutate in the reducer. It's ugly, sure, but it performs better:

 ```js
       ...propNames.reduce(
         (flattenedProps, propName) =>
           Object.assign(flattenedProps, props[propName]),
         {}
       )
```

> 
> If a tree falls in the woods, does it make a sound?
> 
> If a pure function mutates some local data in order to produce an immutable return value, is that ok?
> 
> — Rich Hickey, Clojure